BitwardenShared/UI/Platform/Application/AppProcessor.swift

```diff
@@ -383,14 +383,27 @@ extension AppProcessor {
     /// - Returns: an `AppRoute` if one was successfully built from the URL passed in, `nil` if not.
     ///
     private func getBitwardenUrlRoute(url: URL) async -> AppRoute? {
-        guard let components = URLComponents(url: url, resolvingAgainstBaseURL: true),
-              let scheme = components.scheme,
-              scheme.isBitwardenAppScheme,
-              components.host == "settings",
-              components.path == "/account_security"
-        else { return nil }
-
-        return AppRoute.tab(.settings(.accountSecurity))
+        guard let scheme = url.scheme, scheme.isBitwardenAppScheme else { return nil }
+
+        var route: AppRoute?
+        switch url.absoluteString {
+        case IncomingLinkConstants.accountSecurity:
+            route = AppRoute.tab(.settings(.accountSecurity))
+        case IncomingLinkConstants.authenticatorNewItem:
+            guard let item = await services.authenticatorSyncService?.getTemporaryTotpItem(),
+                  let totpKey = item.totpKey,
+                  let otpAuthModel = OTPAuthModel(otpAuthKey: totpKey) else {
+                coordinator?.showAlert(.defaultAlert(title: Localizations.somethingWentWrong,
+                                                     message: Localizations.pleaseTryAgain))
+                return nil
+            }
+
+            route = AppRoute.tab(.vault(.vaultItemSelection(otpAuthModel)))
+        default:
+            route = nil
+        }
+
+        return route
```

Comment:
🎨 No need for the extra `route` variable, you can simply return on each case:
```suggestion
        switch url.absoluteString {
        case IncomingLinkConstants.accountSecurity:
            return AppRoute.tab(.settings(.accountSecurity))
        case IncomingLinkConstants.authenticatorNewItem:
            guard let item = await services.authenticatorSyncService?.getTemporaryTotpItem(),
                  let totpKey = item.totpKey,
                  let otpAuthModel = OTPAuthModel(otpAuthKey: totpKey) else {
                coordinator?.showAlert(.defaultAlert(title: Localizations.somethingWentWrong,
                                                     message: Localizations.pleaseTryAgain))
                return nil
            }

            return AppRoute.tab(.vault(.vaultItemSelection(otpAuthModel)))
        default:
            return nil
        }
```

Reply:
I like that 👍  Adopted in latest commit.